### PR TITLE
Add FXIOS-7991 [v122] Add custom targeting attribute for review checker being enabled for user

### PR DESCRIPTION
**Pull request by thatswinnie**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7991)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17821)

## :bulb: Description
Adds custom targeting attribute for review checker being enabled for user.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Experiments/Experiments.swift [hunk 1]
```diff
@@ -136,10 +136,14 @@ enum Experiments {
 
         let isPhone = UIDevice.current.userInterfaceIdiom == .phone
 
+        let prefs = UserDefaults(suiteName: AppInfo.sharedContainerIdentifier)!
```
Comment (adudenamedruby) [inline Client/Experiments/Experiments.swift:139]: I am uncomfortable with force unwrapping in prod code (and that swiftlint rule is coming soon). Can we unwrap this?

**Review by adudenamedruby — APPROVED** [review]

**Review by isabelrios — APPROVED** [review]

**Comment by thatswinnie** [comment]
@Mergifyio backport release/v122

File: Client/Experiments/Experiments.swift [hunk 2]
```diff
@@ -136,10 +136,16 @@ enum Experiments {
 
         let isPhone = UIDevice.current.userInterfaceIdiom == .phone
 
+        var isReviewCheckerEnabled = false
+        if let prefs = UserDefaults(suiteName: AppInfo.sharedContainerIdentifier) {
```
Comment (jhugman) [inline Client/Experiments/Experiments.swift:140]: Is the defaults firefox-ios uses for all preferences now?

Comment (lmarceau) [reply Client/Experiments/Experiments.swift:140]: Great question, @thatswinnie left for vacation but I wonder why the `suiteName` was used here. We use the suite name in older part of the code, but I'm unsure if there's any reason why this was used here.

**Pull request merged**